### DISCUSSION TITLE
How to construct run command line arguments conditionally?

### DISCUSSION THREAD

**Discussion by vladak** [community/community#26408] [Actions]
<p>I have a simple matrix build:</p>
<pre><code class="lang-auto">jobs:
  build:
    name: ${{ matrix.os }}
    runs-on: ${{ matrix.os }}
    strategy:
      fail-fast: false
      matrix:
        os: [ubuntu-latest, macos-latest, windows-latest]
    steps:
    - name: Checkout master branch
      uses: actions/checkout@v2
    - name: Unix build
      env:
        PULL_REQUEST: ${{ github.head_ref }}
        REPO_SLUG: ${{ github.repository }}
        REF: ${{ github.ref }}
        COVERALLS_TOKEN: ${{ secrets.COVERALLS_TOKEN }}
        SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      if: runner.os == 'Linux' || runner.os == 'macOS'
      run: ./dev/main
    - name: Windows build
      run: mvn -B -V verify
      if: runner.os == 'Windows'
</code></pre>
<p>The <code>dev/main</code> is a <code>bash</code> script that evaluates the passed environment variables and based on their content constructs a <code>mvn</code> command line.</p>
<p>For instance, if <code>$REPO_SLUG</code> has certain value and <code>$PULL_REQUEST</code> is empty, it will append <code>-P sonar -Dsonar.login=$SONAR_TOKEN sonar:sonar</code> to the arguments and eventually run <code>mvn</code> with these accumulated arguments.</p>
<p>You can see that the Windows build lacks this logic - it runs Maven directly. I’d like all the builds use the same logic however I don’t want to recode the <code>bash</code> script in <code>PowerShell</code>. Is there a way how to make it work solely in the workflow definition ?</p>

**Comment by Simran-B** [comment, marked as answer]
<p>Bash is available on Windows too:</p>
<pre><code class="lang-yaml">    - name: Build
      shell: bash
      env:
        PULL_REQUEST: ${{ github.head_ref }}
        REPO_SLUG: ${{ github.repository }}
        REF: ${{ github.ref }}
        COVERALLS_TOKEN: ${{ secrets.COVERALLS_TOKEN }}
        SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      run: ./dev/main
</code></pre>
<p>Documentation: <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell">Using a specific shell</a></p>

**Comment by vladak** [comment]
<p>works nicely, thanks !</p>
